.github/scripts/setup-env.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+# Prepare conda
+set +x && eval "$($(which conda) shell.bash hook)" && set -x
+
+# Setup the OS_TYPE environment variable that should be used for conditions involving the OS below.
+case $(uname) in
```

Comment:
Are you planning to extend CI to other platforms? Because this file seems a lot more general than what we use it for right now?

Reply:
Maybe, I'm not sure. Leaving it just in case